test/docfx.Test/lsp/LanguageServerTestClient.cs

```diff
@@ -155,15 +162,22 @@ private async Task<ILanguageClient> InitializeClient(string workingDirectory, Pa
                     DocumentChanges = true,
                     FailureHandling = FailureHandlingKind.Undo,
                 })
+                .OnLogMessage(message =>
+                {
+                    if (message.Type == MessageType.Error)
+                    {
+                        _notificationSync.TrySetException(new InvalidOperationException(message.Message));
+                    }
+                })
                 .OnPublishDiagnostics(item =>
                 {
                     _diagnostics[ToFile(item.Uri)] = JToken.FromObject(item.Diagnostics, JsonUtility.Serializer);
                     OnNotification();
                 }));
 
-            await Task.WhenAll(
-                client.Initialize(default),
-                LanguageServerHost.StartLanguageServer(workingDirectory, new(), clientPipe.Reader, serverPipe.Writer, package, this));
+            Task.Run(() => LanguageServerHost.RunLanguageServer(workingDirectory, new(), clientPipe.Reader, serverPipe.Writer, package, this)).GetAwaiter();
+
+            await client.Initialize(default);
```

Comment:
why split this

Reply:
because now the Task returned from `RunLanguageServer` method won't complete until the server is shutdown, hence the rename from `Start` to `Run`